flagevalmm/evaluator/multi_inference_evaluator.py

```diff
@@ -0,0 +1,235 @@
+import json
+from collections import defaultdict
+from typing import Callable, Optional, Dict, List, Tuple, Union
+import os.path as osp
+from flagevalmm.evaluator.base_evaluator import BaseEvaluator
+from flagevalmm.registry import EVALUATORS
+from flagevalmm.common.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+@EVALUATORS.register_module()
+class MultiInferenceEvaluator(BaseEvaluator):
+    """
+    Multi-inference evaluator for evaluating the accuracy of model's multiple inference results.
+    """
+
+    def __init__(
+        self,
+        is_clean: bool = True,
+        use_llm_evaluator: bool = False,
+        eval_func: Optional[Union[Callable, str]] = None,
+        base_dir: str = "",
+        detailed_keys: Optional[List[str]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            is_clean=is_clean,
+            use_llm_evaluator=use_llm_evaluator,
+            eval_func=eval_func,
+            base_dir=base_dir,
+            detailed_keys=detailed_keys,
+            **kwargs,
+        )
+
+    def expand_multi_inference_predictions(
+        self, predictions: List[Dict]
+    ) -> Tuple[List[Dict], Dict]:
+        """
+        Expand multiple inference predictions into individual predictions.
+
+        Returns:
+            expanded_predictions: List of individual predictions
+            question_mapping: Mapping from expanded prediction index to original question info
+        """
+        expanded_predictions = []
+        question_mapping = {}
```

Comment:
Recommand to use `dataclass` instead of Dict
```
@dataclass
class QuestionMapping:
    original_question_id: str
    is_multi_inference: bool
    inference_index: int
    total_inferences: int
```